includes/multi-currency/MultiCurrency.php

```diff
@@ -871,17 +865,15 @@ public function get_raw_conversion( float $amount, string $to_currency, string $
 		$from_currency = strtoupper( $from_currency );
 		foreach ( [ $to_currency, $from_currency ] as $code ) {
 			if ( ! isset( $enabled_currencies[ $code ] ) ) {
-				$message = 'Invalid currency passed to get_raw_conversion: ' . $code;
-				Logger::error( $message );
-				throw new InvalidCurrencyException( $message, 'wcpay_multi_currency_invalid_currency', 500 );
+				$this->log_and_throw_invalid_currency_exception( __FUNCTION__, $code );
 			}
 		}
 
 		// Get the rates.
 		$to_currency_rate   = $enabled_currencies[ $to_currency ]->get_rate();
 		$from_currency_rate = $enabled_currencies[ $from_currency ]->get_rate();
 
-		// Thrown an exception in case from_currency_rate is less than or equal to 0.
+		// Throw an exception in case from_currency_rate is less than or equal to 0.
 		if ( 0 >= $from_currency_rate ) {
 			$message = 'Invalid rate for from_currency in get_raw_conversion: ' . $from_currency_rate;
```

Comment:
I think you missed this one.

Reply:
Not sure how I missed this comment before, it wasn't here when I replied previously.

> I think what @fashuvo meant at first was removing the duplicate code in this method when throwing errors. Maybe you can convert it to a more generic one that also accepts rate errors?

My reply above kind of covers this. If we were to create a handler for throwing the `rate` exception, then we would need to change the messages thrown, unless we allowed to pass messages in. Doing that wouldn't reduce any code or clarify anything, unless I am misunderstanding.

Reply:
Gotcha. I was also fine with the previous approach, so maybe you can merge this one, and let it be handled by some other issue, if it's needed. The only thing that's concerning is the dynamicity of the error log, but I guess that was an issue for the server code, not the client code, right? You can merge this IMO.